_posts/2023-01-19-blog-new-svg-engine.md

```diff
@@ -63,7 +63,7 @@ The LBSE effort began in October 2019 as **a research project**, to find out an
 
 After an extensive phase of experimentation, two abandoned approaches, and a long time spent on regression fixing, the LBSE prototype was finally finished after almost two years of work. It passed all 60k+ WebKit layout tests and offered initial support for compositing, 3D transformations, `z-index`, and more. The intent was to prove that we can reach feature parity with the legacy SVG engine and retrieve the very same visual results, pixel-by-pixel (except for progressions of LBSE). Shortly after the finalization, the prototype was presented during the <a href="https://blogs.igalia.com/nzimmermann/posts/2021-10-13-svg-performance/">WebKit contributors meeting in 2021</a>.
 
-As the name "prototype" indicates, LBSE was not ready for integration into WebKit at this point. It **replaced** the old SVG engine with a new one, resulting in a monolithic patch exceeding *650 kb* of code changes. WebKit demands small patches, with ChangeLogs, tests, etc. -- no WebKit reviewer would sign off a patch replacing a core component of WebCore in one shot. Splitting up into small pieces is also not going to work, since SVG needs to be kept intact upstream all the time. Duplicating the whole SVG engine? Not practicable either. With that problem in mind, a fruitful discussion took place with Apple during and after the WebKit contributors meeting: a realistic upstreaming strategy was defined - thanks *Simon Fraser* for suggesting a pragmatic approach!
+As the name "prototype" indicates, LBSE was not ready for integration into WebKit at this point. It **replaced** the old SVG engine with a new one, resulting in a monolithic patch exceeding *650 KB* of code changes. External contributions generally demand small patches, with ChangeLogs, tests, etc. -- no conscientious reviewer in any company would approve a patch replacing a core component of a browser engine in one shot. Splitting up into small pieces is also not going to work, since SVG needs to be kept intact upstream all the time. Duplicating the whole SVG engine? Not practicable either. With that problem in mind, a fruitful discussion took place with Apple during and after the WebKit contributors meeting: a realistic upstreaming strategy was defined - thanks *Simon Fraser* for suggesting a pragmatic approach!
```

Comment:
It's not only _external_ contributions the one that should be in self-contained patches with tests. That's true for _all_ contributions to the WebKit project 😉 

Also I wouldn't put “in any company” because there can be community projects which are _not_ driven by any company with similar strict contribution guidelines. Maybe something like “no conscientious reviewer in a sizeable project would approve [...]”, WDYT?

Reply:
Sounds great